ambry-clustermap/src/test/java/com.github.ambry.clustermap/HelixParticipantTest.java

```diff
@@ -63,19 +65,97 @@
 public class HelixParticipantTest {
   private final MockHelixManagerFactory helixManagerFactory;
   private final Properties props;
+  private final String clusterName = "HelixParticipantTestCluster";
 
   public HelixParticipantTest() throws Exception {
     List<com.github.ambry.utils.TestUtils.ZkInfo> zkInfoList = new ArrayList<>();
     zkInfoList.add(new com.github.ambry.utils.TestUtils.ZkInfo(null, "DC0", 2199, false));
     JSONObject zkJson = constructZkLayoutJSON(zkInfoList);
     props = new Properties();
     props.setProperty("clustermap.host.name", "localhost");
-    props.setProperty("clustermap.cluster.name", "HelixParticipantTestCluster");
+    props.setProperty("clustermap.cluster.name", clusterName);
     props.setProperty("clustermap.datacenter.name", "DC0");
     props.setProperty("clustermap.dcs.zk.connect.strings", zkJson.toString(2));
     helixManagerFactory = new MockHelixManagerFactory();
   }
 
+  private ReplicaId createMockReplicaId(String partitionIdString) {
+    ReplicaId replicaId = Mockito.mock(ReplicaId.class);
+    PartitionId partitionId = Mockito.mock(PartitionId.class);
+    Mockito.when(partitionId.toPathString()).thenReturn(partitionIdString);
+    Mockito.when(replicaId.getPartitionId()).thenReturn(partitionId);
+    return replicaId;
+  }
+
+  private void listIsExpectedSize(List list, int expectedSize, String name) {
```

Comment:
we usually write some token documentation for private functions too. For example, `name` is not a self explanatory argument here i.e. I dunno how the function is supposed to/will use it.

Reply:
Also try to keep all the functions decorated as `@Test` in the beginning and the private functions at the bottom before the mock classes. IMO, this helps with readability because I get to see all the actual tests in one place and at the top. The private functions are just helpers.

Reply:
I agree that name "name" isn't descriptive enough and should be changed to "listName" to make it more descriptive, but I'm not sure that adding token documentation is necessary or useful (hence "token").  Check out  the "Bad Comments" section of Chapter 4 of Robert Martin's "Clean Code": http://ricardogeek.com/docs/clean_code.html